src/Compilers/CSharp/Portable/Parser/AbstractLexer.cs

```diff
@@ -131,9 +130,18 @@ protected XmlSyntaxDiagnosticInfo MakeError(int position, int width, XmlParseErr
 
         private int GetLexemeOffsetFromPosition(int position)
         {
-            return position >= TextWindow.LexemeStartPosition ? position - TextWindow.LexemeStartPosition : position;
+            return position >= LexemeStartPosition ? position - LexemeStartPosition : position;
         }
 
+        protected string GetNonInternedLexemeText()
+            => TextWindow.GetText(LexemeStartPosition, intern: false);
+
+        protected string GetInternedLexemeText()
+            => TextWindow.GetText(LexemeStartPosition, intern: true);
```

Comment:
these .GetText helpers on the TextWindow were used all over the place.  MOved to two simple helpers on Lexer for getting either an interned or non-interned chunk of text for the current lexeme.